### ISSUE TITLE
Problem when batteries are not charged synchronously

### ISSUE THREAD

**Issue by ckvsoft** [pulquero/BatteryAggregator#18]
Today I noticed a problem - I have 2 serial batteries that can each be charged with 50A

Battery Aggregator shows me the correct sum of 100A

For some reason the two batteries were not switched to charging synchronously - then an attempt was made to charge the first of the batteries with the 100A - this caused the BMS to switch off - then an attempt was made to charge the second battery with 100A and it was also switched off by the BMS

This continued alternately

lg Chris

**Comment by pulquero** [comment]
Please can you give complete details about your setup. Are you using DVCC?

**Comment by ckvsoft** [comment]
Yes
3x Multiplus II 5000
2x jkbms


![grafik](https://github.com/pulquero/BatteryAggregator/assets/16374447/26cf9a43-860c-4131-9a01-6ba3e346d542)
![grafik](https://github.com/pulquero/BatteryAggregator/assets/16374447/78a68e22-ee46-4a21-8e61-a2d995bc7043)

**Comment by pulquero** [comment]
Can you also include dbus-spy of the batteries and multiplus. How are the batteries wired to multiplus so they can be charged independently? If we can find some sync charging flag in dbus, then I can use that to adjust the charging current reported by battery aggregator.

**Comment by ckvsoft** [comment]
now it's getting difficult. I have no idea what to do with it.
The battery packs have their own BMS and are connected in parallel to the Victron via a strip.
If the serial battery regulates down, the aggregator does it correctly and divides it up. But if only one of them means that it should be charged then the amps, the 2x 50A, are delivered to a battery.

**Comment by ckvsoft** [comment]
If serial battery regulates the value for one of the two packs down, this also works, only if one of the two does not go into charging mode then the full amps are charged into a battery pack. Which then has a high voltage error and switches off. Of course it could also be that the error is in the serial battery

**Comment by ckvsoft** [comment]
This is one battery pack


![WhatsApp Image 2024-02-15 at 10 01 49 (1)](https://github.com/pulquero/BatteryAggregator/assets/16374447/b68674d6-57c4-4adc-ae43-7ca25deb1bf3)
![WhatsApp Image 2024-02-15 at 10 01 49 (2)](https://github.com/pulquero/BatteryAggregator/assets/16374447/14ba22e8-7ad1-4b5f-a31b-c327284c2984)
![WhatsApp Image 2024-02-15 at 10 01 49 (3)](https://github.com/pulquero/BatteryAggregator/assets/16374447/7b853ea1-a16e-4e12-9b82-8da5e4fd08b0)
![WhatsApp Image 2024-02-15 at 10 01 49 (4)](https://github.com/pulquero/BatteryAggregator/assets/16374447/beaf8ceb-b067-41bd-aaab-d0b66764c3d5)
![WhatsApp Image 2024-02-15 at 10 01 49 (5)](https://github.com/pulquero/BatteryAggregator/assets/16374447/3b9613ad-c84a-4b73-b5df-1a388e8bed08)
![WhatsApp Image 2024-02-15 at 10 01 49 (6)](https://github.com/pulquero/BatteryAggregator/assets/16374447/7c79b8ec-21f0-4add-b17c-f3bf159fd96b)
![WhatsApp Image 2024-02-1![WhatsApp Image 2024-02-15 at 10 01 49](https://github.com/pulquero/BatteryAggregator/assets/16374447/f7ff3fe6-424b-4d02-aa84-9bc39d0adaf2)
5 at 10 01 49 (7)](https://github.com/pulquero/BatteryAggregator/assets/16374447/5e51c5d2-ecf5-46f4-8070-c6028a252ad3)
![WhatsApp Image 2024-02-15 at 10 01 49 (8)](https://github.com/pulquero/BatteryAggregator/assets/16374447/0bb70826-fdde-42c2-918a-8d4780affd97)
![WhatsApp Image 2024-02-15 at 10 01 49](https://github.com/pulquero/BatteryAggregator/assets/16374447/f7ff3fe6-424b-4d02-aa84-9bc39d0adaf2)

**Comment by pulquero** [comment]
Thanks. I think I need to add support for Io/AllowToCharge and that should hopefully fix it.

**Comment by pulquero** [comment]
Released a new version that should fix it, please test.

**Comment by ckvsoft** [comment]
Thanks for the quick processing. I will test it. But it can take some time because it is not always the case that the batteries have different charging cycles.
I'll get in touch again tomorrow

**Comment by ckvsoft** [comment]
So far there is no error. Thanks again, hoping that was it.

**Comment by SergioRius** [comment]
> Released a new version that should fix it, please test.

I have a problem where one of the batteries enters float before the other and then that last one doesn't get charged anymore. The problem is that the driver get the CVL from the floating battery.
Do you think that new version could also solve it?

**Comment by pulquero** [comment]
No, this sounds like a new separate issue. Probably need to be smarter with how the maxchargevoltage is being aggregated. New issue plus screenshots of this would be helpful.

**Comment by Nurmi78** [comment]
Hi

have the similar issue like ckvsoft and already installed newest version - but noticed the same "deadlock behaviour" again:

my setup is also 3x Multiplus II 5000 with 2 battery packs with 2x jkbms (max charge current = 55A) and Louisvdw/dbus-serialbattery driver installed where I set MAX_BATTERY_CHARGE_CURRENT to 50A

aggregator shows correct CCL 100A

but then this happens:

![WhatsApp Image 2024-02-24 at 12 47 47](https://github.com/pulquero/BatteryAggregator/assets/161292061/b4387e72-1039-4d7d-bdbc-6b4cf8444de5)

as you can see batteries are not charged equally and once one battery hits 55Amps BMS stops charging and in the same second full 100A hits the second BMS which also goes down

you think you can help me with that?

**Comment by Nurmi78** [comment]
max charge current of each battery is linear with max of 50A regarding the linear mode of dbus-serialbattery driver - so when both batteries are allowed to charge with 50A aggregator shows 100A - when driver puts the limit down for some reasons like cell voltage, temperature,... CCL in aggregator seems to be right for me

at the moment it looks like this:
battery 1
![1](https://github.com/pulquero/BatteryAggregator/assets/161292061/1bdb484a-da87-4886-89d8-1c699d1361c8)
battery2
![2](https://github.com/pulquero/BatteryAggregator/assets/161292061/f2931ac1-0eca-4fd1-ae50-b6e8ed227891)
aggregator
![3](https://github.com/pulquero/BatteryAggregator/assets/161292061/18891480-759c-492c-9c46-8627c0b331af)

as far as I can see aggregator always aggregates right from the batteries but I cant say if there is a delay because I have to go through several clicks to see parameters of batteries and aggregator

so the timing might be an issue because when one BMS goes down I can see way too much Amps going to second BMS

what I wonder about is why aggregator overrides the 50Amps charge limit of dbus-serialbattery driver - I think thats the beginning of the issue

**Comment by Nurmi78** [comment]
no you can see in screenshot of my first posting what I mean with "override" - aggregator says CCL is 97.770A but first battery is charged with 53.90A although CCL of battery is set to 50A in driver

in this situation two days ago I had CCL of 50A in both batteries and CCL of 100A in aggregator

I have no plan where the "unequal" charging of 53.90A and 43.87A comes from

short after my screenshots charging of the first battery passes 55A and so BMS goes down

**Comment by pulquero** [comment]
Ah gotcha. So maybe something like:
sum the charging currents, and scale to reduce any overcurrent, e.g. 97.77 * 50/53.9.
I can try adding that in addition to scaling back during balancing.

**Comment by pulquero** [comment]
New version released.

**Comment by Nurmi78** [comment]
thanks I already installed it
unfortunately this behaviour doesnt occur regularly so Im heading back to you once I know more

**Comment by Nurmi78** [comment]
Hi

so far I didnt met all the conditions to load with full 100A where the problem occurs in the past

but today i noticed another strange thing...

CCL of first battery is 5A (due to wrong SOC delivered from BMS)
CCL of second battery is 42A (due to temperature limitation from serialbattery driver)

but CCL from aggregator says 23A ? how can this happen? in my opinion ist should aggregate to 47Amps, right?

**Comment by pulquero** [comment]
This maybe caused by the new scale-back functionality. I've added more logging so you should know why.

**Comment by Nurmi78** [comment]
log says:

@4000000065e4915105165094 INFO:battery:Max charge current is 49.0 but scaling back to 19.276160503540517 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.5420000000000003)
@4000000065e4915113f8e464 INFO:battery:Max charge current is 49.0 but scaling back to 19.276160503540517 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.5420000000000003)
@4000000065e4915122e17f2c INFO:battery:Max charge current is 49.0 but scaling back to 19.276160503540517 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.5420000000000003)
@4000000065e4915131d79afc INFO:battery:Max charge current is 49.0 but scaling back to 19.276160503540517 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.5420000000000003)
@4000000065e4915205356984 INFO:battery:Max charge current is 49.0 but scaling back to 19.276160503540517 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.5420000000000003)
@4000000065e49152142c5844 INFO:battery:Max charge current is 49.0 but scaling back to 19.276160503540517 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.5420000000000003)
@4000000065e49152231740e4 INFO:battery:Max charge current is 49.0 but scaling back to 19.276160503540517 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.5420000000000003)
@4000000065e49152320cf724 INFO:battery:Max charge current is 49.0 but scaling back to 18.992248062015502 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.58)
@4000000065e49153057c2eb4 INFO:battery:Max charge current is 49.0 but scaling back to 18.992248062015502 as limit exceeded for battery com.victronenergy.battery.ttyUSB1 (overcurrent ratio: 2.58)

**Comment by pulquero** [comment]
That would suggest a ccl of 49A is causing about 13A on battery 1, so it has scaled to 19A to keep battery 1 within the 5A ccl. I've just added even more logging to make this even clearer.

**Comment by Nurmi78** [comment]
in thias moment it happended again that aggregator tried to charge one battery with over 150 !! Amps although CCL of battery is 48A

log file without date or time is neot very useful to me ;)

here it is:

@4000000065e9ac7c2865935c INFO:battery:Max charge current is 79.167 but scaling back to 77.11339853892216 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0266309292546782)
@4000000065e9ac7c2865b29c INFO:battery:Battery currents:
@4000000065e9ac7c2865be54 com.victronenergy.battery.ttyUSB0: 41.75, limit 40.667
@4000000065e9ac7c2865ca0c com.victronenergy.battery.ttyUSB1: 34.29, limit 38.5
@4000000065e9ac7c2865d5c4 
@4000000065e9ac7d28c0a7c4 INFO:battery:Max charge current is 79.167 but scaling back to 76.4178587467363 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.035975114958074)
@4000000065e9ac7d28dc3e44 INFO:battery:Battery currents:
@4000000065e9ac7d28dc4de4 com.victronenergy.battery.ttyUSB0: 42.13, limit 40.667
@4000000065e9ac7d28dc5d84 com.victronenergy.battery.ttyUSB1: 33.72, limit 38.5
@4000000065e9ac7d28dc693c 
@4000000065e9ac7e28843334 INFO:battery:Max charge current is 79.167 but scaling back to 77.46593813763234 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0219588364029804)
@4000000065e9ac7e288e9b44 INFO:battery:Battery currents:
@4000000065e9ac7e288eaae4 com.victronenergy.battery.ttyUSB0: 41.56, limit 40.667
@4000000065e9ac7e288eba84 com.victronenergy.battery.ttyUSB1: 33.72, limit 38.5
@4000000065e9ac7e288ec63c 
@4000000065e9ac8028d3223c INFO:battery:Max charge current is 79.167 but scaling back to 77.82171595358957 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0172867435512822)
@4000000065e9ac8028de7894 INFO:battery:Battery currents:
@4000000065e9ac8028de8834 com.victronenergy.battery.ttyUSB0: 41.37, limit 40.667
@4000000065e9ac8028de97d4 com.victronenergy.battery.ttyUSB1: 33.53, limit 38.5
@4000000065e9ac8028dea38c 
@4000000065e9ac81293297bc INFO:battery:Max charge current is 79.167 but scaling back to 77.46593813763234 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0219588364029804)
@4000000065e9ac81293b2b0c INFO:battery:Battery currents:
@4000000065e9ac81293b3aac com.victronenergy.battery.ttyUSB0: 41.56, limit 40.667
@4000000065e9ac81293b4a4c com.victronenergy.battery.ttyUSB1: 33.33, limit 38.5
@4000000065e9ac81293b5604 
@4000000065e9ac82292443c4 INFO:battery:Max charge current is 79.167 but scaling back to 78.54316635764822 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0079425578478864)
@4000000065e9ac82292d773c INFO:battery:Battery currents:
@4000000065e9ac82292d8ac4 com.victronenergy.battery.ttyUSB0: 40.99, limit 40.667
@4000000065e9ac82292d967c com.victronenergy.battery.ttyUSB1: 33.33, limit 38.5
@4000000065e9ac82292da234 
@4000000065e9ac8329590f64 INFO:battery:Max charge current is 79.167 but scaling back to 76.76405314735337 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0313030221063761)
@4000000065e9ac832990bd4c INFO:battery:Battery currents:
@4000000065e9ac832990d0d4 com.victronenergy.battery.ttyUSB0: 41.94, limit 40.667
@4000000065e9ac832990dc8c com.victronenergy.battery.ttyUSB1: 33.33, limit 38.5
@4000000065e9ac832990e844 
@4000000065e9ac8529d7e424 INFO:battery:Max charge current is 79.167 but scaling back to 76.4178587467363 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.035975114958074)
@4000000065e9ac8529dcc23c INFO:battery:Battery currents:
@4000000065e9ac8529dcd1dc com.victronenergy.battery.ttyUSB0: 42.13, limit 40.667
@4000000065e9ac8529dce17c com.victronenergy.battery.ttyUSB1: 34.1, limit 38.5
@4000000065e9ac8529dced34 
@4000000065e9ac862a0df3fc INFO:battery:Max charge current is 79.167 but scaling back to 77.11339853892216 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0266309292546782)
@4000000065e9ac862a12a334 INFO:battery:Battery currents:
@4000000065e9ac862a12b2d4 com.victronenergy.battery.ttyUSB0: 41.75, limit 40.667
@4000000065e9ac862a12c274 com.victronenergy.battery.ttyUSB1: 33.91, limit 38.5
@4000000065e9ac862a12ce2c 
@4000000065e9ac872a792d34 INFO:battery:Max charge current is 79.167 but scaling back to 75.7347539167255 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0453193006614698)
@4000000065e9ac872a794c74 INFO:battery:Battery currents:
@4000000065e9ac872a79582c com.victronenergy.battery.ttyUSB0: 42.51, limit 40.667
@4000000065e9ac872a7963e4 com.victronenergy.battery.ttyUSB1: 33.91, limit 38.5
@4000000065e9ac872a796f9c 
@4000000065e9ac892ae2b8bc INFO:battery:Max charge current is 79.167 but scaling back to 77.46593813763234 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0219588364029804)
@4000000065e9ac892ae82b44 INFO:battery:Battery currents:
@4000000065e9ac892ae83ae4 com.victronenergy.battery.ttyUSB0: 41.56, limit 40.667
@4000000065e9ac892ae8469c com.victronenergy.battery.ttyUSB1: 33.72, limit 38.5
@4000000065e9ac892ae85254 
@4000000065e9ac8b2b3009b4 INFO:battery:Max charge current is 79.167 but scaling back to 78.54316635764822 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0079425578478864)
@4000000065e9ac8b2b37c244 INFO:battery:Battery currents:
@4000000065e9ac8b2b37d1e4 com.victronenergy.battery.ttyUSB0: 40.99, limit 40.667
@4000000065e9ac8b2b37dd9c com.victronenergy.battery.ttyUSB1: 33.72, limit 38.5
@4000000065e9ac8b2b37ed3c 
@4000000065e9ac8d2b77097c INFO:battery:Max charge current is 79.167 but scaling back to 78.90893110294118 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0032704649961885)
@4000000065e9ac8d2b7c704c INFO:battery:Battery currents:
@4000000065e9ac8d2b7c7c04 com.victronenergy.battery.ttyUSB0: 40.8, limit 40.667
@4000000065e9ac8d2b7c8ba4 com.victronenergy.battery.ttyUSB1: 33.33, limit 38.5
@4000000065e9ac8d2b7c975c 
@4000000065e9ac902c02ea14 INFO:battery:Max charge current is 79.167 but scaling back to 77.82171595358957 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0172867435512822)
@4000000065e9ac902c0ac9b4 INFO:battery:Battery currents:
@4000000065e9ac902c0ad954 com.victronenergy.battery.ttyUSB0: 41.37, limit 40.667
@4000000065e9ac902c0ae8f4 com.victronenergy.battery.ttyUSB1: 33.14, limit 38.5
@4000000065e9ac902c0af4ac 
@4000000065e9ac922c4cace4 INFO:battery:Max charge current is 79.167 but scaling back to 76.4178587467363 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.035975114958074)
@4000000065e9ac922c51aa3c INFO:battery:Battery currents:
@4000000065e9ac922c51b5f4 com.victronenergy.battery.ttyUSB0: 42.13, limit 40.667
@4000000065e9ac922c51c594 com.victronenergy.battery.ttyUSB1: 33.91, limit 38.5
@4000000065e9ac922c51d14c 
@4000000065e9ac932c8bcd0c INFO:battery:Max charge current is 79.167 but scaling back to 77.46593813763234 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0219588364029804)
@4000000065e9ac932ca252ac INFO:battery:Battery currents:
@4000000065e9ac932ca26634 com.victronenergy.battery.ttyUSB0: 41.56, limit 40.667
@4000000065e9ac932ca271ec com.victronenergy.battery.ttyUSB1: 33.91, limit 38.5
@4000000065e9ac932ca27da4 
@4000000065e9ac942c9d4d84 INFO:battery:Max charge current is 79.167 but scaling back to 77.11339853892216 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0266309292546782)
@4000000065e9ac942ca50614 INFO:battery:Battery currents:
@4000000065e9ac942ca515b4 com.victronenergy.battery.ttyUSB0: 41.75, limit 40.667
@4000000065e9ac942ca5216c com.victronenergy.battery.ttyUSB1: 33.91, limit 38.5
@4000000065e9ac942ca5310c 
@4000000065e9ac962d1c3d9c INFO:battery:Max charge current is 79.167 but scaling back to 76.07477289697543 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.040647207809772)
@4000000065e9ac962d21564c INFO:battery:Battery currents:
@4000000065e9ac962d2165ec com.victronenergy.battery.ttyUSB0: 42.32, limit 40.667
@4000000065e9ac962d2171a4 com.victronenergy.battery.ttyUSB1: 34.29, limit 38.5
@4000000065e9ac962d217d5c 
@4000000065e9ac972d569b04 INFO:battery:Max charge current is 79.167 but scaling back to 76.4178587467363 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.035975114958074)
@4000000065e9ac972d5f3a0c INFO:battery:Battery currents:
@4000000065e9ac972d5f49ac com.victronenergy.battery.ttyUSB0: 42.13, limit 40.667
@4000000065e9ac972d5f5564 com.victronenergy.battery.ttyUSB1: 34.29, limit 38.5
@4000000065e9ac972d5f611c 
@4000000065e9ac982d86a3e4 INFO:battery:Max charge current is 79.167 but scaling back to 76.76405314735337 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0313030221063761)
@4000000065e9ac982d8bf72c INFO:battery:Battery currents:
@4000000065e9ac982d8c06cc com.victronenergy.battery.ttyUSB0: 41.94, limit 40.667
@4000000065e9ac982d8c1284 com.victronenergy.battery.ttyUSB1: 34.29, limit 38.5
@4000000065e9ac982d8c2224 
@4000000065e9ac9b2e331084 INFO:battery:Max charge current is 79.167 but scaling back to 77.82171595358957 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0172867435512822)
@4000000065e9ac9b2e3cfb94 INFO:battery:Battery currents:
@4000000065e9ac9b2e3d074c com.victronenergy.battery.ttyUSB0: 41.37, limit 40.667
@4000000065e9ac9b2e3d16ec com.victronenergy.battery.ttyUSB1: 33.72, limit 38.5
@4000000065e9ac9b2e3d22a4 
@4000000065ea4fc20d227704 BatteryAggregator/setup: ++ Installing Battery Aggregator service
@4000000065ea4fc20ea07414 BatteryAggregator/setup: ++ Install Python library velib_python (master) from https://github.com/pulquero/velib_python into /data/BatteryAggregator/ext
@4000000065ea4fc300aa1504 BatteryAggregator/setup: restarting BatteryAggregator service
@4000000065ea4fc30108abdc INFO:battery:Exit
@4000000065ea4fc30a92ac34 BatteryAggregator/setup: restarting BatteryAggregator logger
@4000000065ea4fc30f06ded4 BatteryAggregator/setup: restarting generator service
@4000000065ea4fc31175527c BatteryAggregator/setup: completed
@4000000065ea4fc4066b70b4 INFO:dbusmonitor:===== Search on dbus for services that we will monitor starting... =====
@4000000065ea4fc4069d65c4 INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB0, scanning and storing items
@4000000065ea4fc406de2014 INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB0 has device instance 2
@4000000065ea4fc408cca1d4 INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB1, scanning and storing items
@4000000065ea4fc412a74a9c INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB1 has device instance 1
@4000000065ea4fc414a6e35c INFO:dbusmonitor:===== Search on dbus for services that we will monitor finished =====
@4000000065ea4fc50662e534 INFO:battery:Waiting for batteries (attempt 1 of 30)...
@4000000065ea4fc50694f1b4 INFO:root:registered ourselves on D-Bus as com.victronenergy.battery.aggregator
@4000000065ea4fc50847dd8c INFO:battery:Registered Battery Aggregator com.victronenergy.battery.aggregator
@4000000065eabadf3a3c35f4 INFO:battery:Max charge current is 94.0 but scaling back to 92.51589091654705 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0160416666666667)
@4000000065eabadf3a412794 INFO:battery:Battery currents:
@4000000065eabadf3a41334c com.victronenergy.battery.ttyUSB0: 48.77, limit 48.0
@4000000065eabadf3a4142ec com.victronenergy.battery.ttyUSB1: -67.25, limit 46.0
@4000000065eabadf3a414ea4 
@4000000065eabb9c0f46c634 INFO:battery:Max charge current is 94.0 but scaling back to 80.31327874688502 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.1704166666666667)
@4000000065eabb9c0f4f78c4 INFO:battery:Battery currents:
@4000000065eabb9c0f4f8864 com.victronenergy.battery.ttyUSB0: 56.18, limit 48.0
@4000000065eabb9c0f4f941c com.victronenergy.battery.ttyUSB1: -41.38, limit 46.0
@4000000065eabb9c0f4fa3bc 
@4000000065eabca618601eb4 INFO:battery:Max charge current is 94.0 but scaling back to 68.70717222476016 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.368125)
@4000000065eabca61863d7d4 INFO:battery:Battery currents:
@4000000065eabca61863e774 com.victronenergy.battery.ttyUSB0: 65.67, limit 48.0
@4000000065eabca61863f32c com.victronenergy.battery.ttyUSB1: -82.38, limit 46.0
@4000000065eabca61863fee4 
@4000000065eabca818f22584 INFO:battery:Max charge current is 94.0 but scaling back to 81.41465175027066 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.1545833333333333)
@4000000065eabca818fd83ac INFO:battery:Battery currents:
@4000000065eabca818fd9734 com.victronenergy.battery.ttyUSB0: 55.42, limit 48.0
@4000000065eabca818fda2ec com.victronenergy.battery.ttyUSB1: -69.55, limit 46.0
@4000000065eabca818fdaea4 
@4000000065eabca919218a54 INFO:battery:Max charge current is 94.0 but scaling back to 93.98042074567797 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0002083333333334)
@4000000065eabca91926780c INFO:battery:Battery currents:
@4000000065eabca9192687ac com.victronenergy.battery.ttyUSB0: 48.01, limit 48.0
@4000000065eabca919269364 com.victronenergy.battery.ttyUSB1: -61.88, limit 46.0
@4000000065eabca91926a304 
@4000000065eabdfc340c8044 INFO:battery:Max charge current is 94.0 but scaling back to 61.909989023051594 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.5183333333333333)
@4000000065eabdfc34131bac INFO:battery:Battery currents:
@4000000065eabdfc34132f34 com.victronenergy.battery.ttyUSB0: 72.88, limit 48.0
@4000000065eabdfc34133aec com.victronenergy.battery.ttyUSB1: -102.88, limit 46.0
@4000000065eabdfc341346a4 
@4000000065eabdfe3473547c INFO:battery:Max charge current is 94.0 but scaling back to 72.70383499838866 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.2929166666666667)
@4000000065eabdfe34784dec INFO:battery:Battery currents:
@4000000065eabdfe34785d8c com.victronenergy.battery.ttyUSB0: 62.06, limit 48.0
@4000000065eabdfe34786944 com.victronenergy.battery.ttyUSB1: -82.57, limit 46.0
@4000000065eabdfe347878e4 
@4000000065eabdff34929094 INFO:battery:Max charge current is 94.0 but scaling back to 82.83458784652102 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.1347916666666666)
@4000000065eabdff349cbe0c INFO:battery:Battery currents:
@4000000065eabdff349ccdac com.victronenergy.battery.ttyUSB0: 54.47, limit 48.0
@4000000065eabdff349cd964 com.victronenergy.battery.ttyUSB1: -72.23, limit 46.0
@4000000065eabdff349ce51c 
@4000000065eabfd029f8126c INFO:battery:Max charge current is 94.0 but scaling back to 90.05988023952096 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.04375)
@4000000065eabfd029f831ac INFO:battery:Battery currents:
@4000000065eabfd029f83d64 com.victronenergy.battery.ttyUSB0: 50.1, limit 48.0
@4000000065eabfd029f84d04 com.victronenergy.battery.ttyUSB1: -0.0, limit 46.0
@4000000065eabfd029f858bc 
@4000000065eabfd129f91054 INFO:battery:Max charge current is 94.0 but scaling back to 52.71643883631265 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.783125)
@4000000065eabfd12a0218d4 INFO:battery:Battery currents:
@4000000065eabfd12a022874 com.victronenergy.battery.ttyUSB0: 85.59, limit 48.0
@4000000065eabfd12a023814 com.victronenergy.battery.ttyUSB1: -105.38, limit 46.0
@4000000065eabfd12a0243cc 
@4000000065eabfd32a62364c INFO:battery:Max charge current is 94.0 but scaling back to 60.80043120873198 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.5460416666666665)
@4000000065eabfd32a640ef4 INFO:battery:Battery currents:
@4000000065eabfd32a641aac com.victronenergy.battery.ttyUSB0: 74.21, limit 48.0
@4000000065eabfd32a642664 com.victronenergy.battery.ttyUSB1: -88.9, limit 46.0
@4000000065eabfd32a64321c 
@4000000065eabfd42aa1140c INFO:battery:Max charge current is 94.0 but scaling back to 69.10706080563638 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.3602083333333335)
@4000000065eabfd42aa1334c INFO:battery:Battery currents:
@4000000065eabfd42aa13f04 com.victronenergy.battery.ttyUSB0: 65.29, limit 48.0
@4000000065eabfd42aa14ea4 com.victronenergy.battery.ttyUSB1: -77.59, limit 46.0
@4000000065eabfd42aa15a5c 
@4000000065eabfd52aad83a4 INFO:battery:Max charge current is 94.0 but scaling back to 76.94406548431105 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.2216666666666667)
@4000000065eabfd52ac2bd3c INFO:battery:Battery currents:
@4000000065eabfd52ac2d0c4 com.victronenergy.battery.ttyUSB0: 58.64, limit 48.0
@4000000065eabfd52ac2dc7c com.victronenergy.battery.ttyUSB1: -70.89, limit 46.0
@4000000065eabfd52ac2e834 
@4000000065eac2381d64027c INFO:battery:Max charge current is 94.0 but scaling back to 63.397498946185195 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.4827083333333333)
@4000000065eac2381d68fbec INFO:battery:Battery currents:
@4000000065eac2381d690b8c com.victronenergy.battery.ttyUSB0: 71.17, limit 48.0
@4000000065eac2381d691b2c com.victronenergy.battery.ttyUSB1: -46.75, limit 46.0
@4000000065eac2381d6926e4 
@4000000065eac23a1dd4e8d4 INFO:battery:Max charge current is 94.0 but scaling back to 73.15175097276266 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.285)
@4000000065eac23a1dd9edfc INFO:battery:Battery currents:
@4000000065eac23a1dd9fd9c com.victronenergy.battery.ttyUSB0: 61.68, limit 48.0
@4000000065eac23a1dda0d3c com.victronenergy.battery.ttyUSB1: -66.29, limit 46.0
@4000000065eac23a1dda18f4 
@4000000065eac23b1e12f3a4 INFO:battery:Max charge current is 94.0 but scaling back to 80.86021505376345 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.1624999999999999)
@4000000065eac23b1e1312e4 INFO:battery:Battery currents:
@4000000065eac23b1e131ab4 com.victronenergy.battery.ttyUSB0: 55.8, limit 48.0
@4000000065eac23b1e132a54 com.victronenergy.battery.ttyUSB1: -66.29, limit 46.0
@4000000065eac23b1e13360c 
@4000000065eac23c1e1ed69c INFO:battery:Max charge current is 94.0 but scaling back to 88.38393731635652 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0635416666666666)
@4000000065eac23c1e274e94 INFO:battery:Battery currents:
@4000000065eac23c1e275e34 com.victronenergy.battery.ttyUSB0: 51.05, limit 48.0
@4000000065eac23c1e276dd4 com.victronenergy.battery.ttyUSB1: -59.39, limit 46.0
@4000000065eac23c1e27798c 
@4000000065eac4580510f57c INFO:battery:Max charge current is 94.0 but scaling back to 75.71740224869944 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.2414583333333333)
@4000000065eac458052b5b4c INFO:battery:Battery currents:
@4000000065eac458052b6ed4 com.victronenergy.battery.ttyUSB0: 59.59, limit 48.0
@4000000065eac458052b7a8c com.victronenergy.battery.ttyUSB1: -64.76, limit 46.0
@4000000065eac458052b8644 
@4000000065eac4590616960c INFO:battery:Max charge current is 94.0 but scaling back to 83.71057513914657 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.1229166666666666)
@4000000065eac4590616b54c INFO:battery:Battery currents:
@4000000065eac4590616c104 com.victronenergy.battery.ttyUSB0: 53.9, limit 48.0
@4000000065eac4590616ccbc com.victronenergy.battery.ttyUSB1: -56.52, limit 46.0
@4000000065eac4590616d874 
@4000000065eac45a057721bc INFO:battery:Max charge current is 94.0 but scaling back to 89.71962616822431 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0477083333333332)
@4000000065eac45a059e0e94 INFO:battery:Battery currents:
@4000000065eac45a059e221c com.victronenergy.battery.ttyUSB0: 50.29, limit 48.0
@4000000065eac45a059e31bc com.victronenergy.battery.ttyUSB1: -50.58, limit 46.0
@4000000065eac45a059e3d74 
@4000000065eac67b376f854c INFO:battery:Max charge current is 94.0 but scaling back to 92.51589091654705 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0160416666666667)
@4000000065eac67b3777748c INFO:battery:Battery currents:
@4000000065eac67b3777842c com.victronenergy.battery.ttyUSB0: 48.77, limit 48.0
@4000000065eac67b377793cc com.victronenergy.battery.ttyUSB1: -0.0, limit 46.0
@4000000065eac67b37779f84 
@4000000065ead3c5230e4804 INFO:battery:Max charge current is 92.0 but scaling back to 85.54823711739635 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.0754166666666667)
@4000000065ead3c52313fcf4 INFO:battery:Battery currents:
@4000000065ead3c5231408ac com.victronenergy.battery.ttyUSB0: 51.62, limit 48.0
@4000000065ead3c52314184c com.victronenergy.battery.ttyUSB1: -0.0, limit 44.0
@4000000065ead3c523142404 
@4000000065ead3c7241b81e4 INFO:battery:Max charge current is 92.0 but scaling back to 52.87989462339839 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.7397916666666668)
@4000000065ead3c724451284 INFO:battery:Battery currents:
@4000000065ead3c72445260c com.victronenergy.battery.ttyUSB0: 83.51, limit 48.0
@4000000065ead3c7244531c4 com.victronenergy.battery.ttyUSB1: -0.0, limit 44.0
@4000000065ead3c724453d7c 
@4000000065ead3c8254e490c INFO:battery:Max charge current is 92.0 but scaling back to 49.192380528016045 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.8702083333333333)
@4000000065ead3c82554a20c INFO:battery:Battery currents:
@4000000065ead3c82554b1ac com.victronenergy.battery.ttyUSB0: 89.77, limit 48.0
@4000000065ead3c82554bd64 com.victronenergy.battery.ttyUSB1: -0.0, limit 44.0
@4000000065ead3c82554c91c 
@4000000065ead3c925753db4 INFO:battery:Max charge current is 92.0 but scaling back to 52.64036237930623 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.7477083333333334)
@4000000065ead3c9257e55d4 INFO:battery:Battery currents:
@4000000065ead3c9257e6574 com.victronenergy.battery.ttyUSB0: 83.89, limit 48.0
@4000000065ead3c9257e7514 com.victronenergy.battery.ttyUSB1: -0.0, limit 44.0
@4000000065ead3c9257e80cc 
@4000000065ead4082252c804 INFO:battery:Max charge current is 92.0 but scaling back to 48.98502495840266 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 1.878125)
@4000000065ead408226a47a4 INFO:battery:Battery currents:
@4000000065ead408226a5f14 com.victronenergy.battery.ttyUSB0: 90.15, limit 48.0
@4000000065ead408226a6acc com.victronenergy.battery.ttyUSB1: 40.04, limit 44.0
@4000000065ead408226a7684 
@4000000065ead40922785d1c INFO:battery:Max charge current is 92.0 but scaling back to 28.976377952755904 as limit exceeded for battery com.victronenergy.battery.ttyUSB0 (overcurrent ratio: 3.1750000000000003)
@4000000065ead409228c5a4c INFO:battery:Battery currents:
@4000000065ead409228c6dd4 com.victronenergy.battery.ttyUSB0: 152.4, limit 48.0
@4000000065ead409228c798c com.victronenergy.battery.ttyUSB1: -98.09, limit 44.0
@4000000065ead409228c8544

**Comment by Nurmi78** [comment]
as far as i can see its same behaviour we had at begining...battery CCL says 46A but battery charging Amps goes up to 50-52A

still dont know where this comes from :/

**Comment by pulquero** [comment]
@4000000065ead409228c5a4c INFO:battery:Battery currents:
@4000000065ead409228c6dd4 com.victronenergy.battery.ttyUSB0: 152.4, limit 48.0
@4000000065ead409228c798c com.victronenergy.battery.ttyUSB1: -98.09, limit 44.0

Looks like there is an added complication of one battery charging, while the other is discharging. If it is discharging, is /Io/AllowToCharge still 1?

**Comment by Nurmi78** [comment]
I can take a look next time it happens but Im pretty sure this parameter is OK

you see the CCLs of 48 and 44 - aggregator said a correct CCL of 92 but for some reason after ttyUSB0 battery came online again immediatley all current from PV plus all possible currecnt from other battery (DCL = 100Amps) goes tom this battery forcing BMS to disconnect again

while ttyUSB0 was disconnected I noticed that ttyUSB1 was not charged with fully possible 44A - charging was floating between 27-38 Amps which is kind of strange too

I did a uninstall and fresh install of aggregator - after that charging was back to normal behaviour

**Comment by Nurmi78** [comment]
have another strange behaviour right now...after 4 hours of charging aggregator suddenly says "not connected" and in log is also says no batteries discoverd although both batteries are still there in Venus OS:

@4000000065ec510015b267bc INFO:battery:Battery currents:
@4000000065ec510015b27b44 com.victronenergy.battery.ttyUSB2: 38.12, limit 35.833
@4000000065ec510015b28ae4 com.victronenergy.battery.ttyUSB1: 30.36, limit 38.167
@4000000065ec510015b2969c 
@4000000065ec51011560407c INFO:battery:Max charge current is 74.0 but scaling back to 68.85593352376006 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0747076716992716)
@4000000065ec51011565321c INFO:battery:Battery currents:
@4000000065ec5101156541bc com.victronenergy.battery.ttyUSB2: 38.51, limit 35.833
@4000000065ec510115654d74 com.victronenergy.battery.ttyUSB1: 30.36, limit 38.167
@4000000065ec51011565592c 
@4000000065ec510315a6ee3c INFO:battery:Max charge current is 74.0 but scaling back to 70.26078431372548 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0532190997125557)
@4000000065ec510315abdfdc INFO:battery:Battery currents:
@4000000065ec510315abef7c com.victronenergy.battery.ttyUSB2: 37.74, limit 35.833
@4000000065ec510315abff1c com.victronenergy.battery.ttyUSB1: 30.36, limit 38.167
@4000000065ec510315ac0ad4 
@4000000065ec510415d35954 INFO:battery:Max charge current is 74.166 but scaling back to 71.13464341541756 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0426143499009293)
@4000000065ec510415dad364 INFO:battery:Battery currents:
@4000000065ec510415dae304 com.victronenergy.battery.ttyUSB2: 37.36, limit 35.833
@4000000065ec510415daeebc com.victronenergy.battery.ttyUSB1: 29.22, limit 38.333
@4000000065ec510415dafa74 
@4000000065ec510515f6a094 INFO:battery:Max charge current is 74.166 but scaling back to 70.77470780292943 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0479167248067423)
@4000000065ec51051643c694 INFO:battery:Battery currents:
@4000000065ec51051643da1c com.victronenergy.battery.ttyUSB2: 37.55, limit 35.833
@4000000065ec51051643e5d4 com.victronenergy.battery.ttyUSB1: 29.22, limit 38.333
@4000000065ec51051643f18c 
@4000000065ec51071655fabc INFO:battery:Max charge current is 74.166 but scaling back to 71.49825875706215 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0373119749951163)
@4000000065ec5107165ad104 INFO:battery:Battery currents:
@4000000065ec5107165adcbc com.victronenergy.battery.ttyUSB2: 37.17, limit 35.833
@4000000065ec5107165aec5c com.victronenergy.battery.ttyUSB1: 29.03, limit 38.333
@4000000065ec5107165af814 
@4000000065ec5109169bf4cc INFO:battery:Max charge current is 74.166 but scaling back to 71.13464341541756 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0426143499009293)
@4000000065ec510916a0cefc INFO:battery:Battery currents:
@4000000065ec510916a0de9c com.victronenergy.battery.ttyUSB2: 37.36, limit 35.833
@4000000065ec510916a0ee3c com.victronenergy.battery.ttyUSB1: 28.84, limit 38.333
@4000000065ec510916a0f9f4 
@4000000065ec510b16eb4d4c INFO:battery:Max charge current is 74.166 but scaling back to 70.06565457421566 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0585214746183687)
@4000000065ec510b16f33c8c INFO:battery:Battery currents:
@4000000065ec510b16f34c2c com.victronenergy.battery.ttyUSB2: 37.93, limit 35.833
@4000000065ec510b16f35bcc com.victronenergy.battery.ttyUSB1: 29.79, limit 38.333
@4000000065ec510b16f36784 
@4000000065ec510d174607dc INFO:battery:Max charge current is 74.166 but scaling back to 70.41839634340221 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0532190997125557)
@4000000065ec510d174d7e04 INFO:battery:Battery currents:
@4000000065ec510d174d8da4 com.victronenergy.battery.ttyUSB2: 37.74, limit 35.833
@4000000065ec510d174d995c com.victronenergy.battery.ttyUSB1: 29.6, limit 38.333
@4000000065ec510d174da514 
@4000000065ec511117dee894 INFO:battery:Max charge current is 74.166 but scaling back to 69.71642911857292 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0638238495241816)
@4000000065ec511117e3e9d4 INFO:battery:Battery currents:
@4000000065ec511117e3f974 com.victronenergy.battery.ttyUSB2: 38.12, limit 35.833
@4000000065ec511117e40914 com.victronenergy.battery.ttyUSB1: 30.17, limit 38.333
@4000000065ec511117e414cc 
@4000000065ec51121811a8c4 INFO:battery:Max charge current is 74.166 but scaling back to 70.06565457421566 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0585214746183687)
@4000000065ec511218167b24 INFO:battery:Battery currents:
@4000000065ec511218168ac4 com.victronenergy.battery.ttyUSB2: 37.93, limit 35.833
@4000000065ec51121816967c com.victronenergy.battery.ttyUSB1: 30.17, limit 38.333
@4000000065ec51121816a61c 
@4000000065ec51141873e534 INFO:battery:Max charge current is 74.166 but scaling back to 69.71642911857292 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0638238495241816)
@4000000065ec51141878e28c INFO:battery:Battery currents:
@4000000065ec51141878f614 com.victronenergy.battery.ttyUSB2: 38.12, limit 35.833
@4000000065ec5114187901cc com.victronenergy.battery.ttyUSB1: 30.36, limit 38.333
@4000000065ec511418790d84 
@4000000065ec511618c8c7ac INFO:battery:Max charge current is 74.166 but scaling back to 70.41839634340221 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0532190997125557)
@4000000065ec5116191ed304 INFO:battery:Battery currents:
@4000000065ec5116191ee68c com.victronenergy.battery.ttyUSB2: 37.74, limit 35.833
@4000000065ec5116191ef244 com.victronenergy.battery.ttyUSB1: 30.36, limit 38.333
@4000000065ec5116191efdfc 
@4000000065ec51181919319c INFO:battery:Max charge current is 74.166 but scaling back to 69.71642911857292 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0638238495241816)
@4000000065ec5118191e139c INFO:battery:Battery currents:
@4000000065ec5118191e233c com.victronenergy.battery.ttyUSB2: 38.12, limit 35.833
@4000000065ec5118191e2ef4 com.victronenergy.battery.ttyUSB1: 30.17, limit 38.333
@4000000065ec5118191e3aac 
@4000000065ec51191941b4b4 INFO:battery:Max charge current is 74.166 but scaling back to 70.06565457421566 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0585214746183687)
@4000000065ec51191946ae24 INFO:battery:Battery currents:
@4000000065ec51191946bdc4 com.victronenergy.battery.ttyUSB2: 37.93, limit 35.833
@4000000065ec51191946c97c com.victronenergy.battery.ttyUSB1: 29.6, limit 38.333
@4000000065ec51191946d91c 
@4000000065ec511a1962311c INFO:battery:Max charge current is 74.166 but scaling back to 70.41839634340221 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0532190997125557)
@4000000065ec511a19671704 INFO:battery:Battery currents:
@4000000065ec511a196726a4 com.victronenergy.battery.ttyUSB2: 37.74, limit 35.833
@4000000065ec511a19673644 com.victronenergy.battery.ttyUSB1: 29.79, limit 38.333
@4000000065ec511a196741fc 
@4000000065ec511c19bfd1dc INFO:battery:Max charge current is 74.166 but scaling back to 69.71642911857292 as limit exceeded for battery com.victronenergy.battery.ttyUSB2 (overcurrent ratio: 1.0638238495241816)
@4000000065ec511c19c58e9c INFO:battery:Battery currents:
@4000000065ec511c19c5a224 com.victronenergy.battery.ttyUSB2: 38.12, limit 35.833
@4000000065ec511c19c5addc com.victronenergy.battery.ttyUSB1: 30.55, limit 38.333
@4000000065ec511c19c5b994 
@4000000065ec511c31dd9e0c BatteryAggregator/setup: ++ Installing Battery Aggregator service
@4000000065ec511c334b8dbc BatteryAggregator/setup: ++ Install Python library velib_python (master) from https://github.com/pulquero/velib_python into /data/BatteryAggregator/ext
@4000000065ec511d2a5c7d74 BatteryAggregator/setup: restarting BatteryAggregator service
@4000000065ec511d2abc2d8c INFO:battery:Exit
@4000000065ec511d333dfd14 BatteryAggregator/setup: restarting BatteryAggregator logger
@4000000065ec511d3866492c BatteryAggregator/setup: restarting generator service
@4000000065ec511d3a2f1694 BatteryAggregator/setup: completed
@4000000065ec511e2a851fcc INFO:main:Starting...
@4000000065ec511e2b9497e4 INFO:dbusmonitor:===== Search on dbus for services that we will monitor starting... =====
@4000000065ec511e2ba9ba0c INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB2, scanning and storing items
@4000000065ec511e2bd8ec14 INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB2 has device instance 1
@4000000065ec511e3016dffc INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB1, scanning and storing items
@4000000065ec511e303f95dc INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB1 has device instance 2
@4000000065ec511e3224ff7c INFO:dbusmonitor:===== Search on dbus for services that we will monitor finished =====
@4000000065ec511f242db224 INFO:main:Waiting for batteries (attempt 1 of 30)...
@4000000065ec51202423656c INFO:main:Waiting for batteries (attempt 2 of 30)...
@4000000065ec512124331124 INFO:main:Waiting for batteries (attempt 3 of 30)...
@4000000065ec5122242b2d9c INFO:main:Waiting for batteries (attempt 4 of 30)...
@4000000065ec5123242d6fbc INFO:main:Waiting for batteries (attempt 5 of 30)...
@4000000065ec5124242428bc INFO:main:Waiting for batteries (attempt 6 of 30)...
@4000000065ec5125242db9f4 INFO:main:Waiting for batteries (attempt 7 of 30)...
@4000000065ec51262436783c INFO:main:Waiting for batteries (attempt 8 of 30)...
@4000000065ec5127243ae8f4 INFO:main:Waiting for batteries (attempt 9 of 30)...
@4000000065ec512824338654 INFO:main:Waiting for batteries (attempt 10 of 30)...
@4000000065ec5129244b6b84 INFO:main:Waiting for batteries (attempt 11 of 30)...
@4000000065ec512a2421ea84 INFO:main:Waiting for batteries (attempt 12 of 30)...
@4000000065ec512b2444149c INFO:main:Waiting for batteries (attempt 13 of 30)...
@4000000065ec512c242ceaec INFO:main:Waiting for batteries (attempt 14 of 30)...
@4000000065ec512d242307ac INFO:main:Waiting for batteries (attempt 15 of 30)...
@4000000065ec512e2422c92c INFO:main:Waiting for batteries (attempt 16 of 30)...
@4000000065ec512f242866ac INFO:main:Waiting for batteries (attempt 17 of 30)...
@4000000065ec513024263bfc INFO:main:Waiting for batteries (attempt 18 of 30)...
@4000000065ec5131242de4ec INFO:main:Waiting for batteries (attempt 19 of 30)...
@4000000065ec513224287e1c INFO:main:Waiting for batteries (attempt 20 of 30)...
@4000000065ec5133242b7fa4 INFO:main:Waiting for batteries (attempt 21 of 30)...
@4000000065ec5134242bbe24 INFO:main:Waiting for batteries (attempt 22 of 30)...
@4000000065ec5135242db60c INFO:main:Waiting for batteries (attempt 23 of 30)...
@4000000065ec51362432b74c INFO:main:Waiting for batteries (attempt 24 of 30)...
@4000000065ec51372429222c INFO:main:Waiting for batteries (attempt 25 of 30)...
@4000000065ec51382430e674 INFO:main:Waiting for batteries (attempt 26 of 30)...
@4000000065ec513924272e2c INFO:main:Waiting for batteries (attempt 27 of 30)...
@4000000065ec513a242cbff4 INFO:main:Waiting for batteries (attempt 28 of 30)...
@4000000065ec513b242c373c INFO:main:Waiting for batteries (attempt 29 of 30)...
@4000000065ec513c242a6a4c INFO:main:Waiting for batteries (attempt 30 of 30)...
@4000000065ec513c2431f7e4 WARNING:main:No batteries discovered!
@4000000065ec513c24399cec INFO:main:Exit
@4000000065ec513d177fbd4c INFO:main:Starting...
@4000000065ec513d1888a5b4 INFO:dbusmonitor:===== Search on dbus for services that we will monitor starting... =====
@4000000065ec513d18a355bc INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB2, scanning and storing items
@4000000065ec513d18cbcd1c INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB2 has device instance 1
@4000000065ec513d1ac6167c INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB1, scanning and storing items
@4000000065ec513d1aecd08c INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB1 has device instance 2
@4000000065ec513d1d1e856c INFO:dbusmonitor:===== Search on dbus for services that we will monitor finished =====
@4000000065ec513e2428a52c INFO:main:Waiting for batteries (attempt 1 of 30)...
@4000000065ec513f24234a14 INFO:main:Waiting for batteries (attempt 2 of 30)...
@4000000065ec5140242b9afc INFO:main:Waiting for batteries (attempt 3 of 30)...
@4000000065ec514124269da4 INFO:main:Waiting for batteries (attempt 4 of 30)...
@4000000065ec5142242be14c INFO:main:Waiting for batteries (attempt 5 of 30)...
@4000000065ec514324309854 INFO:main:Waiting for batteries (attempt 6 of 30)...
@4000000065ec5144242c3354 INFO:main:Waiting for batteries (attempt 7 of 30)...
@4000000065ec51452425b344 INFO:main:Waiting for batteries (attempt 8 of 30)...
@4000000065ec5146242983d4 INFO:main:Waiting for batteries (attempt 9 of 30)...
@4000000065ec51472428ef64 INFO:main:Waiting for batteries (attempt 10 of 30)...
@4000000065ec51482431ccec INFO:main:Waiting for batteries (attempt 11 of 30)...
@4000000065ec5149242acfdc INFO:main:Waiting for batteries (attempt 12 of 30)...
@4000000065ec514a24234244 INFO:main:Waiting for batteries (attempt 13 of 30)...
@4000000065ec514b242aa0fc INFO:main:Waiting for batteries (attempt 14 of 30)...
@4000000065ec514c24216d84 INFO:main:Waiting for batteries (attempt 15 of 30)...
@4000000065ec514d242e1f84 INFO:main:Waiting for batteries (attempt 16 of 30)...
@4000000065ec514e2423d2cc INFO:main:Waiting for batteries (attempt 17 of 30)...
@4000000065ec514f2428acfc INFO:main:Waiting for batteries (attempt 18 of 30)...
@4000000065ec5150242c0474 INFO:main:Waiting for batteries (attempt 19 of 30)...
@4000000065ec51512429ca24 INFO:main:Waiting for batteries (attempt 20 of 30)...
@4000000065ec5152242b1dfc INFO:main:Waiting for batteries (attempt 21 of 30)...
@4000000065ec515324239c1c INFO:main:Waiting for batteries (attempt 22 of 30)...
@4000000065ec515424240594 INFO:main:Waiting for batteries (attempt 23 of 30)...
@4000000065ec51552429c254 INFO:main:Waiting for batteries (attempt 24 of 30)...
@4000000065ec5156242cbc0c INFO:main:Waiting for batteries (attempt 25 of 30)...
@4000000065ec5157242545e4 INFO:main:Waiting for batteries (attempt 26 of 30)...
@4000000065ec51582423368c INFO:main:Waiting for batteries (attempt 27 of 30)...
@4000000065ec51592428b4cc INFO:main:Waiting for batteries (attempt 28 of 30)...
@4000000065ec515a2439faac INFO:main:Waiting for batteries (attempt 29 of 30)...
@4000000065ec515b2424867c INFO:main:Waiting for batteries (attempt 30 of 30)...
@4000000065ec515b242c0474 WARNING:main:No batteries discovered!
@4000000065ec515b243372cc INFO:main:Exit
@4000000065ec515c17a7afdc INFO:main:Starting...
@4000000065ec515c1a2415d4 INFO:dbusmonitor:===== Search on dbus for services that we will monitor starting... =====
@4000000065ec515c1a3fcb94 INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB2, scanning and storing items
@4000000065ec515c1a6623fc INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB2 has device instance 1
@4000000065ec515c1c60176c INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB1, scanning and storing items
@4000000065ec515c1c8867bc INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB1 has device instance 2
@4000000065ec515c1e57cca4 INFO:dbusmonitor:===== Search on dbus for services that we will monitor finished =====
@4000000065ec515d242b1dfc INFO:main:Waiting for batteries (attempt 1 of 30)...
@4000000065ec515e2425c2e4 INFO:main:Waiting for batteries (attempt 2 of 30)...
@4000000065ec515f242a00d4 INFO:main:Waiting for batteries (attempt 3 of 30)...
@4000000065ec51602427d23c INFO:main:Waiting for batteries (attempt 4 of 30)...
@4000000065ec5161242ab86c INFO:main:Waiting for batteries (attempt 5 of 30)...
@4000000065ec516224215614 INFO:main:Waiting for batteries (attempt 6 of 30)...
@4000000065ec51632431193c INFO:main:Waiting for batteries (attempt 7 of 30)...
@4000000065ec5164242614ec INFO:main:Waiting for batteries (attempt 8 of 30)...
@4000000065ec516524249dec INFO:main:Waiting for batteries (attempt 9 of 30)...
@4000000065ec5166242e7574 INFO:main:Waiting for batteries (attempt 10 of 30)...
@4000000065ec5167242a52dc INFO:main:Waiting for batteries (attempt 11 of 30)...
@4000000065ec5168242e5634 INFO:main:Waiting for batteries (attempt 12 of 30)...
@4000000065ec5169242be91c INFO:main:Waiting for batteries (attempt 13 of 30)...
@4000000065ec516a24296c64 INFO:main:Waiting for batteries (attempt 14 of 30)...
@4000000065ec516b246694a4 INFO:main:Waiting for batteries (attempt 15 of 30)...
@4000000065ec516c242a9d14 INFO:main:Waiting for batteries (attempt 16 of 30)...
@4000000065ec516d2424cccc INFO:main:Waiting for batteries (attempt 17 of 30)...
@4000000065ec516e24278034 INFO:main:Waiting for batteries (attempt 18 of 30)...
@4000000065ec516f2425e60c INFO:main:Waiting for batteries (attempt 19 of 30)...
@4000000065ec5170246037bc INFO:main:Waiting for batteries (attempt 20 of 30)...
@4000000065ec5171242a3b6c INFO:main:Waiting for batteries (attempt 21 of 30)...
@4000000065ec5172242d8efc INFO:main:Waiting for batteries (attempt 22 of 30)...
@4000000065ec51732430270c INFO:main:Waiting for batteries (attempt 23 of 30)...
@4000000065ec51742423f9dc INFO:main:Waiting for batteries (attempt 24 of 30)...
@4000000065ec517524246f0c INFO:main:Waiting for batteries (attempt 25 of 30)...
@4000000065ec51762428476c INFO:main:Waiting for batteries (attempt 26 of 30)...
@4000000065ec5177242a08a4 INFO:main:Waiting for batteries (attempt 27 of 30)...
@4000000065ec5178242159fc INFO:main:Waiting for batteries (attempt 28 of 30)...
@4000000065ec51792423a7d4 INFO:main:Waiting for batteries (attempt 29 of 30)...
@4000000065ec517a2468a014 INFO:main:Waiting for batteries (attempt 30 of 30)...
@4000000065ec517a2468b39c WARNING:main:No batteries discovered!
@4000000065ec517a2468bf54 INFO:main:Exit
@4000000065ec517b18e2990c INFO:main:Starting...
@4000000065ec517b1a133584 INFO:dbusmonitor:===== Search on dbus for services that we will monitor starting... =====
@4000000065ec517b1b57ee6c INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB2, scanning and storing items
@4000000065ec517b1ba40acc INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB2 has device instance 1
@4000000065ec517b1d930d74 INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB1, scanning and storing items
@4000000065ec517b1dbba414 INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB1 has device instance 2
@4000000065ec517b1f8757ac INFO:dbusmonitor:===== Search on dbus for services that we will monitor finished =====
@4000000065ec517c2428eb7c INFO:main:Waiting for batteries (attempt 1 of 30)...
@4000000065ec517d24237cdc INFO:main:Waiting for batteries (attempt 2 of 30)...
@4000000065ec517e242735fc INFO:main:Waiting for batteries (attempt 3 of 30)...
@4000000065ec517f2422e86c INFO:main:Waiting for batteries (attempt 4 of 30)...
@4000000065ec5180242f4c4c INFO:main:Waiting for batteries (attempt 5 of 30)...
@4000000065ec5181242ac80c INFO:main:Waiting for batteries (attempt 6 of 30)...
@4000000065ec518224268a1c INFO:main:Waiting for batteries (attempt 7 of 30)...
@4000000065ec51832425ab74 INFO:main:Waiting for batteries (attempt 8 of 30)...
@4000000065ec5184242b068c INFO:main:Waiting for batteries (attempt 9 of 30)...
@4000000065ec51852424a9a4 INFO:main:Waiting for batteries (attempt 10 of 30)...
@4000000065ec518624267e64 INFO:main:Waiting for batteries (attempt 11 of 30)...
@4000000065ec5187242e7d44 INFO:main:Waiting for batteries (attempt 12 of 30)...
@4000000065ec51882423656c INFO:main:Waiting for batteries (attempt 13 of 30)...
@4000000065ec51892428b8b4 INFO:main:Waiting for batteries (attempt 14 of 30)...
@4000000065ec518a2420e4cc INFO:main:Waiting for batteries (attempt 15 of 30)...
@4000000065ec518b24251ed4 INFO:main:Waiting for batteries (attempt 16 of 30)...
@4000000065ec518c2429416c INFO:main:Waiting for batteries (attempt 17 of 30)...
@4000000065ec518d2425613c INFO:main:Waiting for batteries (attempt 18 of 30)...
@4000000065ec518e2425de3c INFO:main:Waiting for batteries (attempt 19 of 30)...
@4000000065ec518f242baa9c INFO:main:Waiting for batteries (attempt 20 of 30)...
@4000000065ec519024207384 INFO:main:Waiting for batteries (attempt 21 of 30)...
@4000000065ec5191242acfdc INFO:main:Waiting for batteries (attempt 22 of 30)...
@4000000065ec519224253644 INFO:main:Waiting for batteries (attempt 23 of 30)...
@4000000065ec5193242e94b4 INFO:main:Waiting for batteries (attempt 24 of 30)...
@4000000065ec5194242bcdc4 INFO:main:Waiting for batteries (attempt 25 of 30)...
@4000000065ec519524263044 INFO:main:Waiting for batteries (attempt 26 of 30)...
@4000000065ec5196242ad3c4 INFO:main:Waiting for batteries (attempt 27 of 30)...
@4000000065ec5197242b644c INFO:main:Waiting for batteries (attempt 28 of 30)...
@4000000065ec5198242620a4 INFO:main:Waiting for batteries (attempt 29 of 30)...
@4000000065ec51992465d924 INFO:main:Waiting for batteries (attempt 30 of 30)...
@4000000065ec51992465ecac WARNING:main:No batteries discovered!
@4000000065ec51992465fc4c INFO:main:Exit
@4000000065ec519a16f766f4 INFO:main:Starting...
@4000000065ec519a180237a4 INFO:dbusmonitor:===== Search on dbus for services that we will monitor starting... =====
@4000000065ec519a181da714 INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB2, scanning and storing items
@4000000065ec519a1dd16664 INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB2 has device instance 1
@4000000065ec519a1fbff3dc INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB1, scanning and storing items
@4000000065ec519a1fe830a4 INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB1 has device instance 2
@4000000065ec519a21bbf2bc INFO:dbusmonitor:===== Search on dbus for services that we will monitor finished =====
@4000000065ec519b24239c1c INFO:main:Waiting for batteries (attempt 1 of 30)...
@4000000065ec519c24211794 INFO:main:Waiting for batteries (attempt 2 of 30)...
@4000000065ec519d242aa4e4 INFO:main:Waiting for batteries (attempt 3 of 30)...
@4000000065ec519e242d313c INFO:main:Waiting for batteries (attempt 4 of 30)...
@4000000065ec519f24319a24 INFO:main:Waiting for batteries (attempt 5 of 30)...
@4000000065ec51a0242bed04 INFO:main:Waiting for batteries (attempt 6 of 30)...
@4000000065ec51a1242a1c2c INFO:main:Waiting for batteries (attempt 7 of 30)...
@4000000065ec51a22421428c INFO:main:Waiting for batteries (attempt 8 of 30)...
@4000000065ec51a3242735fc INFO:main:Waiting for batteries (attempt 9 of 30)...
@4000000065ec51a4242f8eb4 INFO:main:Waiting for batteries (attempt 10 of 30)...
@4000000065ec51a524285edc INFO:main:Waiting for batteries (attempt 11 of 30)...
@4000000065ec51a6242ed334 INFO:main:Waiting for batteries (attempt 12 of 30)...
@4000000065ec51a7242eb00c INFO:main:Waiting for batteries (attempt 13 of 30)...
@4000000065ec51a8242f9e54 INFO:main:Waiting for batteries (attempt 14 of 30)...
@4000000065ec51a924280504 INFO:main:Waiting for batteries (attempt 15 of 30)...
@4000000065ec51aa24262c5c INFO:main:Waiting for batteries (attempt 16 of 30)...
@4000000065ec51ab242dcd7c INFO:main:Waiting for batteries (attempt 17 of 30)...
@4000000065ec51ac24266ec4 INFO:main:Waiting for batteries (attempt 18 of 30)...

**Comment by Nurmi78** [comment]
I did a complete Raspi reboot now and log files say:

@4000000065ec54382d349be4 INFO:main:Waiting for batteries (attempt 16 of 30)...
@4000000065ec54392d332cb4 INFO:main:Waiting for batteries (attempt 17 of 30)...
@4000000065ec543a2d35f3a4 INFO:main:Waiting for batteries (attempt 18 of 30)...
@4000000065ec543b082c6a34 INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB0, scanning and storing items
@4000000065ec543b2562657c INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB0 has device instance 2
@4000000065ec543b2d39fae4 INFO:main:Waiting for batteries (attempt 19 of 30)...
@4000000065ec543b2dad523c INFO:root:registered ourselves on D-Bus as com.victronenergy.battery.aggregator
@4000000065ec543b314ebfd4 INFO:main:Registered Battery Aggregator com.victronenergy.battery.aggregator
@4000000065ec543c2f8b516c CRITICAL:ve_utils:exit_on_error: there was an exception. Printing stacktrace will be tried and then exit
@4000000065ec543c2f8b70ac Traceback (most recent call last):
@4000000065ec543c2f8b881c   File "/data/BatteryAggregator/ext/ve_utils.py", line 22, in exit_on_error
@4000000065ec543c2f8b9f8c     return func(*args, **kwargs)
@4000000065ec543c2f8baf2c   File "/data/BatteryAggregator/ext/dbusmonitor.py", line 347, in _execute_value_changes
@4000000065ec543c2f8bbecc     self.valueChangedCallback(serviceName, objectPath,
@4000000065ec543c2f8d2a14   File "/data/BatteryAggregator/battery_service.py", line 378, in _battery_value_changed
@4000000065ec543c2f8d456c     self._update_battery_value(dbusServiceName, dbusPath, value)
@4000000065ec543c2f8d60c4   File "/data/BatteryAggregator/battery_service.py", line 392, in _update_battery_value
@4000000065ec543c2f8d7834     self._update_battery_path(dbusServiceName, p, self.service[p])
@4000000065ec543c2f8d87d4 AttributeError: 'BatteryAggregatorService' object has no attribute '_update_battery_path'
@4000000065ec543d2f4a971c INFO:main:Starting...
@4000000065ec543d3070e2f4 INFO:dbusmonitor:===== Search on dbus for services that we will monitor starting... =====
@4000000065ec543d30a06704 INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB0, scanning and storing items
@4000000065ec543d30c592a4 INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB0 has device instance 2
@4000000065ec543d32b2c474 INFO:dbusmonitor:Found: com.victronenergy.battery.ttyUSB1, scanning and storing items
@4000000065ec543d32eadbd4 INFO:dbusmonitor:       com.victronenergy.battery.ttyUSB1 has device instance 1
@4000000065ec543d34c2d024 INFO:dbusmonitor:===== Search on dbus for services that we will monitor finished =====
@4000000065ec543e2d3474d4 INFO:main:Waiting for batteries (attempt 1 of 30)...
@4000000065ec543f2d341714 INFO:main:Waiting for batteries (attempt 2 of 30)...
@4000000065ec54402d319674 INFO:main:Waiting for batteries (attempt 3 of 30)...
@4000000065ec54412d36b30c INFO:main:Waiting for batteries (attempt 4 of 30)...
@4000000065ec54422d2de524 INFO:main:Waiting for batteries (attempt 5 of 30)...
@4000000065ec54432d30c76c INFO:main:Waiting for batteries (attempt 6 of 30)...
@4000000065ec54442d33480c INFO:main:Waiting for batteries (attempt 7 of 30)...
@4000000065ec54452d323e6c INFO:main:Waiting for batteries (attempt 8 of 30)...

**Comment by pulquero** [comment]
I've pushed a new version for you to try. It is now event-based rather than polling so minimal lag to the aggregator reflecting changes to the batteries. And, more actively managing CCL, taking into account battery internal resistances for better accuracy.

**Comment by Nurmi78** [comment]
thanks...is it still version 3.0.14 ?

**Comment by pulquero** [comment]
Yep, I've bumped it to 3.0.15 anyway.
